### PR TITLE
RAIL-3674 AttributeFilterButton filtered selection fix

### PR DESCRIPTION
**Pull request by plavemar**
JIRA: RAIL-3674

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 1]
```diff
@@ -348,12 +348,33 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
         [state.validOptions, state.offset, state.limit, resolvedParentFilters],
     );
 
+    const {
+        error: originalTotalCountError,
+        result: originalTotalCount,
+        status: originalTotalCountStatus,
+    } = useCancelablePromise<number>(
+        {
+            // eslint-disable-next-line
```
Comment (plavemar) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:357]: Please let me know if there is any possible way to specify why I'm disabling eslint. 

The es-lint error was stating that there is exact duplicity for both `totalCount` cancelable promises.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5119/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6808/ : SUCCESS in 14m 37s

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:357]: You can write another comment above this one to describe it

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 2]
```diff
@@ -348,12 +348,33 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
...
+            promise: async () => {
+                return getElementTotalCount(
+                    props.workspace,
+                    props.backend,
+                    getObjRef(currentFilter, props.identifier),
+                    state.searchString,
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:363]: Shouldn't the search string be ignored here?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1720

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1457/ : SUCCESS in 1m 13s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1734/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/520/ : SUCCESS in 13m 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6814/ : SUCCESS in 14m 36s

**Pull request merged**